File: lib/biz/spec-iot/src/main/java/site/addzero/biz/spec/iot/protocol/modbus/ModbusRegisterType.java
package site.addzero.biz.spec.iot.protocol.modbus;

/**
 * Common Modbus register types.
 */
public enum ModbusRegisterType {

    COIL_STATUS("01"),
    INPUT_STATUS("02"),
    HOLDING_REGISTER("03"),
    INPUT_REGISTER("04");

    private final String code;

    ModbusRegisterType(String code) {
        this.code = code;
    }

    public String getCode() {
        return code;
    }
}

File: lib/biz/spec-iot/src/main/kotlin/site/addzero/biz/spec/iot/protocol/modbus/ModbusRegisterType.kt
package site.addzero.biz.spec.iot.protocol.modbus

/**
 * Common Modbus register types.
 */
enum class ModbusRegisterType(
    val code: String,
) {
    COIL_STATUS("01"),
    INPUT_STATUS("02"),
    HOLDING_REGISTER("03"),
    INPUT_REGISTER("04"),
}
